packages/editor/src/components/post-text-editor/index.js

```diff
@@ -65,6 +66,9 @@ export default function PostTextEditor() {
 		}
 	};
 
+	// Ensure changes aren't lost when switching modes using shortcuts.
+	useShortcut( 'core/edit-post/toggle-mode', stopEditing );
```

Comment:
This feels bit hacky, but works better than plain side-effect.

@youknowriad, @ntsekouras I would love to get your opinion on this method.

Reply:
Could something like this work?
```
useEffect( () => {
	refValue.current = value;
}, [ value ] );
useEffect( () => {
	return () => {
		const blocks = parse( refValue.current );
		resetEditorBlocks( blocks );
	};
}, [] );
```

I think by using `useEffect` on umount covers any other case that might exist now(or in the future) besides the shortcut. Also by using a ref to get the new value, we don't need to add dependencies to the `unmount` hook.

What do you think?

Reply:
That would work as well. Thank for the suggestion, Nik.

I've updated the code.

Reply:
> but works better than a plain side-effect.

Hey @Mamaduka, could you clarify this bit? From a correctness standpoint, flushing changes on unmount seems much preferable.

Reply:
GitHub failed to update this page, so I only got the latest comments now and you're free to ignore my question. :)

Reply:
I only noticed the issue because of the failing e2e tests. But Nik's suggestion works better than my initial one since refs can be mitted from deps.

My initial effect looked like the example below, and the cleanup method was running on every value change. If we want to sync only on unmount, we must skip dependencies.

```
useEffect( () => {
	return () => {
                if ( isDirty ) {
		  const blocks = parse( value );
		  resetEditorBlocks( blocks );
                }
	};
}, [ isDirty, value ] );
```